### PR TITLE
Add sql mode format restrictions to expr in ticdc filter

### PR DESCRIPTION
**Pull request by hongyunyan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/16689
- Other reference link(s): https://github.com/pingcap/tiflow/pull/10644

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-filter.md [hunk 1]
```diff
@@ -78,10 +78,10 @@ Description of configuration parameters:
 | drop view     | DDL  | |Matches `drop view` event     |
 
 - `ignore-sql`: the DDL statements to be ignored. This parameter accepts an array of strings, in which you can configure multiple regular expressions. This rule only applies to DDL events.
```
Comment (CharlesCheung96) [inline ticdc/ticdc-filter.md:80]: 是否也可以说明下不推荐使用 `ignore-sql` ？

Comment (hongyunyan) [reply ticdc/ticdc-filter.md:80]: 确实，可以把上面支持的 event 也补齐了

File: ticdc/ticdc-filter.md [hunk 2]
```diff
@@ -76,12 +76,36 @@ Description of configuration parameters:
...
+| drop primary key | DDL  | |Matches `drop primary key` event    |
+| rename index | DDL  | |Matches `rename index` event    |
+| alter index visibility | DDL  | |Matches `alter index visibility` event    |
+| alter ttl info | DDL  | |Matches `alter ttl info` event    |
+| alter ttl remove| DDL  | |Matches `alter table remove ttl info` event    |
+| multi schema change | DDL  | |Matches DDL events that change multiple properties of a table within the same DDL statement |
+
+> **Note:**
+>
+> TiDB's DDL statements support changing multiple attributes of a single table at the same time, such as `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;` This operation will be defined as MultiSchemaChange. If you want to filter out this type of DDL, you need to configure "multi schema change".
```
Comment (lilin90) [inline ticdc/ticdc-filter.md:102]: To keep wording consistent with that in line 98.
```suggestion
> TiDB's DDL statements support changing multiple properties of a single table at the same time, such as `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;`. This operation is defined as MultiSchemaChange. If you want to filter out this type of DDL, you need to configure "multi schema change".
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Comment by lilin90** [comment]
I updated this PR in https://github.com/pingcap/docs/pull/16626/commits/02f86e0e773b91c17ae56ea88a4e573bdb6ee33d according to @qiancai's comments for https://github.com/pingcap/docs-cn/pull/16689.

File: ticdc/ticdc-filter.md [hunk 3]
```diff
@@ -57,31 +57,55 @@ Description of configuration parameters:
...
+    | add column | DDL  | |Matches `add column` event    |
+    | drop column | DDL  | |Matches `drop column` event    |
+    | modify column | DDL  | |Matches `modify column` event    |
+    | set default value | DDL  | |Matches `set default value` event    |
+    | add primary key | DDL  | |Matches `add primary key` event    |
+    | drop primary key | DDL  | |Matches `drop primary key` event    |
+    | rename index | DDL  | |Matches `rename index` event    |
+    | alter index visibility | DDL  | |Matches `alter index visibility` event    |
+    | alter ttl info | DDL  | |Matches `alter ttl info` event    |
+    | alter ttl remove| DDL  | |Matches DDL events that remove all the TTL properties of a table |
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:97]:
```suggestion
    | alter ttl remove| DDL  | |Matches DDL events that remove all TTL attributes of a table |
```

Comment (qiancai) [reply ticdc/ticdc-filter.md:97]: ref: https://docs.pingcap.com/tidb/dev/time-to-live#modify-the-ttl-attribute-of-a-table

image

File: ticdc/ticdc-filter.md [hunk 4]
```diff
@@ -57,31 +57,55 @@ Description of configuration parameters:
...
+    | multi schema change | DDL  | |Matches DDL events that change multiple properties of a table within the same DDL statement |
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:98]:
```suggestion
    | multi schema change | DDL  | |Matches DDL events that change multiple attributes of a table within the same DDL statement |
```

File: ticdc/ticdc-filter.md [hunk 5]
```diff
@@ -57,31 +57,55 @@ Description of configuration parameters:
...
+
+> **Note:**
+>
+> TiDB's DDL statements support changing multiple properties of a single table at the same time, such as `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;`. This operation is defined as MultiSchemaChange. If you want to filter out this type of DDL, you need to configure `"multi schema change"` in `ignore-event`.
+
+- `ignore-sql`: the regular expressions of the DDL statements to be filtered out. This parameter accepts an array of strings, in which you can configure multiple regular expressions. This configuration only applies to DDL events.
+- `ignore-delete-value-expr`: this parameter accepts a SQL expression that follows the default SQL Mode, used to filter out the `DELETE` type of DML events with a specified value.
+- `ignore-insert-value-expr`: this parameter accepts a SQL expression that follows the default SQL Mode, used to filter out the `INSERT` type of DML events with a specified value.
+- `ignore-update-old-value-expr`: this parameter accepts a SQL expression that follows the default SQL Mode, used to filter out the `UPDATE` type of DML events with an old specified value.
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:107]:
```suggestion
- `ignore-update-old-value-expr`: this parameter accepts a SQL expression that follows the default SQL mode, used to filter out the `UPDATE` type of DML events with a specified old value.
```

File: ticdc/ticdc-filter.md [hunk 6]
```diff
@@ -57,31 +57,55 @@ Description of configuration parameters:
...
+- `ignore-update-new-value-expr`: this parameter accepts a SQL expression that follows the default SQL Mode, used to filter out the `UPDATE` DML events with a new specified value.
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:108]:
```suggestion
- `ignore-update-new-value-expr`: this parameter accepts a SQL expression that follows the default SQL mode, used to filter out the `UPDATE` DML events with a specified new value.
```

File: ticdc/ticdc-filter.md [hunk 7]
```diff
@@ -57,31 +57,55 @@ Description of configuration parameters:
...
+    | alter ttl remove| DDL  | |Matches DDL events that remove all the TTL properties of a table |
+    | multi schema change | DDL  | |Matches DDL events that change multiple properties of a table within the same DDL statement |
+
+> **Note:**
+>
+> TiDB's DDL statements support changing multiple properties of a single table at the same time, such as `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;`. This operation is defined as MultiSchemaChange. If you want to filter out this type of DDL, you need to configure `"multi schema change"` in `ignore-event`.
+
+- `ignore-sql`: the regular expressions of the DDL statements to be filtered out. This parameter accepts an array of strings, in which you can configure multiple regular expressions. This configuration only applies to DDL events.
+- `ignore-delete-value-expr`: this parameter accepts a SQL expression that follows the default SQL Mode, used to filter out the `DELETE` type of DML events with a specified value.
+- `ignore-insert-value-expr`: this parameter accepts a SQL expression that follows the default SQL Mode, used to filter out the `INSERT` type of DML events with a specified value.
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:106]:
```suggestion
- `ignore-delete-value-expr`: this parameter accepts a SQL expression that follows the default SQL mode, used to filter out the `DELETE` type of DML events with a specified value.
- `ignore-insert-value-expr`: this parameter accepts a SQL expression that follows the default SQL mode, used to filter out the `INSERT` type of DML events with a specified value.
```

Comment (qiancai) [reply ticdc/ticdc-filter.md:106]: ref: https://docs.pingcap.com/tidb/dev/ticdc-ddl#sql-mode

File: ticdc/ticdc-filter.md [hunk 8]
```diff
@@ -57,31 +57,55 @@ Description of configuration parameters:
...
+    | drop primary key | DDL  | |Matches `drop primary key` event    |
+    | rename index | DDL  | |Matches `rename index` event    |
+    | alter index visibility | DDL  | |Matches `alter index visibility` event    |
+    | alter ttl info | DDL  | |Matches `alter ttl info` event    |
+    | alter ttl remove| DDL  | |Matches DDL events that remove all the TTL properties of a table |
+    | multi schema change | DDL  | |Matches DDL events that change multiple properties of a table within the same DDL statement |
+
+> **Note:**
+>
+> TiDB's DDL statements support changing multiple properties of a single table at the same time, such as `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;`. This operation is defined as MultiSchemaChange. If you want to filter out this type of DDL, you need to configure `"multi schema change"` in `ignore-event`.
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:102]:
```suggestion
    > **Note:**
    >
    > TiDB's DDL statements support changing multiple attributes of a single table at the same time, such as `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;`. This operation is defined as MultiSchemaChange. If you want to filter out this type of DDL, you need to configure `"multi schema change"` in `ignore-event`.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**